robot-core/src/main/java/org/obolibrary/robot/IOHelper.java

```diff
@@ -273,6 +283,34 @@ protected static boolean cleanTDB(String tdbDir) {
     return success;
   }
 
+  /**
+   * Get a set of parsers for a given ontology format.
+   *
+   * @param format ontology file format as string
+   * @return set of OWLParserFactories
+   * @throws IllegalArgumentException on bad format
+   */
+  public static Set<OWLParserFactory> getParsersByFormat(String format) {
+    switch (format) {
+      case "omn":
+        return Sets.newHashSet(new ManchesterOWLSyntaxOntologyParserFactory());
+      case "obo":
+        return Sets.newHashSet(new OBOFormatOWLAPIParserFactory());
+      case "ofn":
+        return Sets.newHashSet(new OWLFunctionalSyntaxOWLParserFactory());
+      case "owx":
+        return Sets.newHashSet(new OWLXMLParserFactory());
+      case "owl":
+        return Sets.newHashSet(new RDFXMLParserFactory(), new RioRDFXMLParserFactory());
+      case "ttl":
+        return Sets.newHashSet(new RioTurtleParserFactory(), new TurtleOntologyParserFactory());
+      case "json":
+        return Sets.newHashSet(new RioJsonParserFactory(), new RioJsonLDParserFactory());
```

Comment:
This should be the OBOGraphs parser (stand-alone) or none at all I think.

TBH, I think the --input-format options should reflect identically the `-f` option. Why would we care about RioJson and not about the other more outlandish parsers like KRSS? I would want my parsing to fail if it try parsing with the exact parsers I specify - not because for some reason the second json parser in the list was successful because of its crappy implementation..

Reply:
I was thinking that we should just divide up the parsers that OWLAPI currently uses, but you're right that if we're adding a new option then we don't have to be constrained in that way.

Reply:
The way you do it for `remove` is IMO the best. We use the same extensions:parser mappings here as in `-f`, but, if we ever so want to (I don't), add a bunch more switches later for compare complex combinations `any-owl` for ofn, owx, rdfxml, KRSSParser for krss etc (analogous to axiomtype option in `--axioms`). I would not bother here - the `-f` should be by far sufficient.

Reply:
I spent quite a bit of time digging around OWLAPI and could not find anything for OBOGraphs. Here's a full list of the parsers in a default manager (`OWLManager.createOWLOntologyManager()`):
* RDFXMLParserFactory
* OWLXMLParserFactory
* OWLFunctionalSyntaxOWLParserFactory
* RioTurtleParserFactory
* ManchesterOWLSyntaxOntologyParserFactory
* RioNQuadsParserFactory
* RioJsonParserFactory
* RioNTriplesParserFactory
* RioTrigParserFactory
* RioBinaryRdfParserFactory
* RioJsonLDParserFactory
* RioN3ParserFactory
* RioRDFXMLParserFactory
* RioTrixParserFactory
* TurtleOntologyParserFactory
* OBOFormatOWLAPIParserFactory
* KRSS2OWLParserFactory
* RioRDFaParserFactory

Unless I'm missing something, I only see the Rio JSON parsers.

Reply:
OBOgraphs parser is not in owlapi..  If it is anywhere, it is in the obographs package..

Reply:
I had checked there as well (https://github.com/geneontology/obographs), but didn't see anything about a parser. The manager currently doesn't use OBOGraphs anyway when loading any JSON files, so it seems we're using one of the Rio parsers when we're loading in any JSON. We just use the `OgJsonGenerator` to write OBO graphs JSON.

The OG reader ([OgJsonGenerator](https://github.com/geneontology/obographs/blob/master/obographs-core/src/main/java/org/geneontology/obographs/core/io/OgJsonReader.java)) doesn't produce an OWLOntology.

I was curious to see what's going on... Interestingly, if you load an OG JSON file into ROBOT and print out the format:
```java
// Load the ontology
OWLOntology loadedOntology = manager.loadOntologyFromOntologyDocument(source, config);

// Check for unparsed triples - get the document format and then the loader metadata
OWLDocumentFormat f = manager.getOntologyFormat(loadedOntology);
```

...it tells you it's RDF/XML. Looks like at some point while loading, it's converted:
```
DEBUG Converting from JSON to RDF
```